src/main/java/com/google/cloud/resourcemanager/ConstraintInfo.java

```diff
@@ -0,0 +1,315 @@
+          new Function<
+              com.google.api.services.cloudresourcemanager.model.Constraint, ConstraintInfo>() {
+            @Override
+            public ConstraintInfo apply(
+                com.google.api.services.cloudresourcemanager.model.Constraint pb) {
+              return ConstraintInfo.fromPb(pb);
+            }
+          };
+  static final Function<
+          ConstraintInfo, com.google.api.services.cloudresourcemanager.model.Constraint>
+      TO_PB_FUNCTION =
+          new Function<
+              ConstraintInfo, com.google.api.services.cloudresourcemanager.model.Constraint>() {
+            @Override
+            public com.google.api.services.cloudresourcemanager.model.Constraint apply(
+                ConstraintInfo constraintInfo) {
+              return constraintInfo.toPb();
+            }
+          };
+
+  private BooleanConstraint booleanConstraint;
+  private String constraintDefault;
+  private String description;
+  private String displayName;
+  private ListConstraint listConstraint;
+  private String name;
+  private Integer version;
+
+  /**
+   * A Constraint that allows or disallows a list of string values, which are configured by an
+   * Organization's policy administrator with a Policy.
+   */
+  public static class ListConstraint implements Serializable {
+
+    private static final long serialVersionUID = -2133042982786959352L;
+
+    private final String suggestedValue;
+    private final Boolean supportsUnder;
+
+    ListConstraint(String suggestedValue, Boolean supportsUnder) {
+      this.suggestedValue = suggestedValue;
+      this.supportsUnder = supportsUnder;
+    }
+
+    public String getSuggestedValue() {
+      return suggestedValue;
+    }
+
+    public Boolean getSupportsUnder() {
+      return supportsUnder;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("suggestedValue", getSuggestedValue())
+          .add("supportsUnder", getSupportsUnder())
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(getSuggestedValue(), getSupportsUnder());
+    }
+
+    @Override
+    public final boolean equals(Object obj) {
+      return obj == this
+          || obj != null
+              && obj.getClass().equals(ListConstraint.class)
+              && Objects.equals(toPb(), ((ListConstraint) obj).toPb());
+    }
+
+    com.google.api.services.cloudresourcemanager.model.ListConstraint toPb() {
+      return new com.google.api.services.cloudresourcemanager.model.ListConstraint()
+          .setSuggestedValue(suggestedValue)
+          .setSupportsUnder(supportsUnder);
+    }
+
+    static ListConstraint fromPb(
+        com.google.api.services.cloudresourcemanager.model.ListConstraint listConstraint) {
+      return new ListConstraint(
+          listConstraint.getSuggestedValue(), listConstraint.getSupportsUnder());
+    }
+  }
+
+  public abstract static class Builder {
```

Comment:
Why the extra abstract class here instead of just a concrete Builder class?

Reply:
Thanks, Removed extra abstract class and move ahead with concrete Builder class